### PR TITLE
Cleanup, polish, e2e tests, dir sync, exec

### PR DESCRIPTION
**Pull request by TomasTomecek**
Professional code barf.

* [x] finish the dir sync
* [x] docs

### CODE REVIEW

File: tests/e2e/test_ironman.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import pytest
+from kubernetes import config, client
+from kubernetes.client import V1DeleteOptions
+from kubernetes.client.rest import ApiException
+
+from generator.deploy_openshift_pod import OpenshiftDeployer, VolumeSpec, MappedDir
+from generator.utils import run_command
+
+SANDBOX_IMAGE = "docker.io/usercont/packit-generator"
```
Comment (phracek) [inline tests/e2e/test_ironman.py]: Please add this into conftest.py. I guess I will need it for GeneratorAPI.

File: tests/e2e/test_ironman.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+POD_NAME = "generator-pod"
+NAMESPACE = "myproject"
+
+
+def build_now():
+    project_root = Path(__file__).parent.parent.parent
+    run_command(["docker", "build", "-t", TEST_IMAGE_NAME, str(project_root)])
+
+
+def deploy(test_name: str):
```
Comment (phracek) [inline tests/e2e/test_ironman.py]: This could be a part of `conftest.py` as well, so I will not duplicate code.

**Comment by TomasTomecek** [comment]
Ready for review.
`=== 12 passed, 33 warnings in 22.73 seconds ===`

File: README.md [hunk 3]
```diff
@@ -1,14 +1,52 @@
...
-Generator is used for deploying one POD from another POD.
-The aim is to separate `packit-service` and generation sources in order
-to be secure.
+Run untrusted code in a sandbox - an OpenShift pod.
 
-## How to use generator
+
+## Usage
+
+The simplest use case is to invoke a command in a new openshift pod:
```
Comment (phracek) [inline README.md]: The prerequisite is to run `oc cluster up`. I guess you have to set up also variables like `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT`, haven't you?

File: generator/deploy_openshift_pod.py [hunk 4]
```diff
@@ -20,206 +20,361 @@
...
+        if the command is not set, sleep process runs in the pod and you can use exec
+        to run commands inside
+
+        :param command: command to run in the pod, if None, run sleep
+        """
+        self.deploy_pod(command=command)
+        logger.info("Sandbox pod is deployed.")
+        return self.get_logs()
+
+    def exec(self, command: List[str]) -> str:
```
Comment (phracek) [inline generator/deploy_openshift_pod.py]: later on, this should be a part of some `generator/api.py` file.

File: generator/deploy_openshift_pod.py [hunk 5]
```diff
@@ -20,206 +20,361 @@
...
+    def copy_path_to_pod(self, map: MappedDir):
+        """ copy local path inside the pod """
+        # Copy /tmp/foo local file to /tmp/bar
+        # in a remote pod in namespace <some-namespace>:
+        #   oc cp /tmp/foo <some-namespace>/<some-pod>:/tmp/bar
+        target = f"{self.k8s_namespace_name}/{self.pod_name}:{map.path}"
+        logger.info(f"copy {map.local_dir} -> {target}")
+        # if you're interested: the way openshift does this is that creates a tarball locally
+        # and streams it via exec into the container to a pod process
+        run_command(["oc", "cp", map.local_dir, target])
```
Comment (phracek) [inline generator/deploy_openshift_pod.py]: I would not call it run_command, I would prefer something like `oc_command`. `run_command` is going to be used by `generator/api.py` for calling from https://github.com/packit-service/packit/blob/master/packit/base_git.py#L228

File: generator/deploy_openshift_pod.py [hunk 6]
```diff
@@ -20,206 +20,361 @@
...
 
-    def deploy_pod(self):
+    def copy_path_from_pod(self, map: MappedDir):
+        """ copy remote path content locally """
+        # Copy /tmp/foo from a remote pod to /tmp/bar locally
+        #   oc cp <some-namespace>/<some-pod>:/tmp/foo /tmp/bar
+        target = f"{self.k8s_namespace_name}/{self.pod_name}:{map.path}"
+        logger.info(f"copy {target} -> {map.local_dir}")
+        run_command(["oc", "cp", target, map.local_dir])
```
Comment (phracek) [inline generator/deploy_openshift_pod.py]: the same here.

File: generator/deploy_openshift_pod.py (hunk 4 above)
Comment (TomasTomecek) [reply generator/deploy_openshift_pod.py]: ideally this class should be the api

File: generator/deploy_openshift_pod.py (hunk 5 above)
Comment (TomasTomecek) [reply generator/deploy_openshift_pod.py]: I'm sorry but I have no idea what you mean.

**Comment by phracek** [comment]
I would prefer to merge it and continue with other work.

File: generator/deploy_openshift_pod.py (hunk 5 above)
Comment (phracek) [reply generator/deploy_openshift_pod.py]: I would prefer to merge it and discuss later on.

**Pull request merged**